src/ui/components/FilterTextField.java

```diff
@@ -102,6 +104,7 @@ private void setup() {
 				confirmEdit();
 			} else if (e.getCode() == KeyCode.ESCAPE) {
 				revertEdit();
+				getParent().requestFocus();
```

Comment:
`setOnCancel` could be used here. Each panel would hook into the escape/cancel event and focus itself or something when it triggers.